internal/host/plugin/repository_host_catalog.go

```diff
@@ -143,6 +146,279 @@ func (r *Repository) CreateCatalog(ctx context.Context, c *HostCatalog, _ ...Opt
+	if currentCatalog.GetVersion() != version {
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.New(ctx, errors.VersionMismatch, op, fmt.Sprintf("catalog version mismatch, want=%d, got=%d", currentCatalog.GetVersion(), version))
+	}
+
+	// Clone the catalog so that we can set fields.
+	newCatalog := currentCatalog.clone()
+	var updateAttributes bool
+	var dbMask, nullFields []string
+	for _, f := range fieldMask {
+		switch {
+		case strings.EqualFold("name", f) && c.Name == "":
+			nullFields = append(nullFields, "name")
+			newCatalog.Name = c.Name
+		case strings.EqualFold("name", f) && c.Name != "":
+			dbMask = append(dbMask, "name")
+			newCatalog.Name = c.Name
+		case strings.EqualFold("description", f) && c.Description == "":
+			nullFields = append(nullFields, "description")
+			newCatalog.Description = c.Description
+		case strings.EqualFold("description", f) && c.Description != "":
+			dbMask = append(dbMask, "description")
+			newCatalog.Description = c.Description
+		case strings.EqualFold("attributes", strings.Split(f, ".")[0]):
+			// Flag attributes for updating. While multiple masks may be
+			// sent, we only need to do this once.
+			updateAttributes = true
+		case strings.EqualFold("secrets", f):
+			// While in a similar format, secrets are passed along
+			// wholesale (for the time being). Don't append this mask
+			// field, as secrets do not have a database entry. Clear the
+			// secrets out of the working set after.
+			newCatalog.Secrets = c.Secrets
+			c.Secrets = nil
+
+		default:
+			return nil, db.NoRowsAffected, db.NoRowsAffected, errors.New(ctx, errors.InvalidFieldMask, op, fmt.Sprintf("invalid field mask: %s", f))
+		}
+	}
+
+	if updateAttributes && c.Attributes != nil {
+		dbMask = append(dbMask, "attributes")
+		newCatalog.Attributes, err = patchstruct.PatchBytes(newCatalog.Attributes, c.Attributes)
+		if err != nil {
+			return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op, errors.WithMsg("error in catalog attribute JSON"))
+		}
+
+		// Patch the working set with the new attributes.
+		c.Attributes = newCatalog.Attributes
+	}
+
+	// Get the plugin client.
+	plgClient, ok := r.plugins[currentCatalog.GetPluginId()]
+	if !ok || plgClient == nil {
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("plugin %q not available", currentCatalog.GetPluginId()))
+	}
+
+	// Convert the catalog values to API protobuf values, which is what
+	// we use for the plugin hook calls.
+	currPlgHc, err := toPluginCatalog(ctx, currentCatalog)
+	if err != nil {
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op)
+	}
+	newPlgHc, err := toPluginCatalog(ctx, newCatalog)
+	if err != nil {
+		return nil, db.NoRowsAffected, db.NoRowsAffected, errors.Wrap(ctx, err, op)
+	}
+
+	// Get the secrets for the host catalog.
+	currentCatalogPersisted, err := r.getPersistedDataForCatalog(ctx, currentCatalog)
```

Comment:
I think you need to update your client.  Last week the persisted data for a catalog was changed to be returned with getCatalog.

Reply:
Yep, merged most recent changes yesterday (which led to #1661).